include/RAJA/policy/cuda/forall.hpp

```diff
@@ -74,22 +74,40 @@
 
 #include "RAJA/index/IndexSet.hpp"
 
+#include <algorithm>
+
 namespace RAJA
 {
 
+namespace cuda
+{
+
 namespace impl
 {
-//
-//////////////////////////////////////////////////////////////////////
-//
-// CUDA kernel templates.
-//
-//////////////////////////////////////////////////////////////////////
-//
 
-// INTERNAL namespace to encapsulate helper functions
-namespace INTERNAL
+/*!
+ ******************************************************************************
+ *
+ * \brief calculate gridDim from length of iteration and blockDim
+ *
+ ******************************************************************************
+ */
+RAJA_INLINE
+dim3 getGridDim(size_t len, dim3 blockDim)
```

Comment:
Why this rather than cudaOccupancyMaxPotentialBlockSize or cudaOccupancyMaxPotentialBlockSizeVariableSMem?